.devcontainer/Dockerfile.dev

```diff
@@ -0,0 +1,25 @@
+ARG VARIANT=3.9-bullseye
+FROM mcr.microsoft.com/vscode/devcontainers/python:0-${VARIANT}
+
+ENV VIRTUAL_ENV=/opt/.venv
+ENV FILE_LOCATION "/usr/src/app"
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+ENV LOCAL_USER="vscode"
+
+USER $LOCAL_USER
+
+COPY . $FILE_LOCATION
+
+WORKDIR $FILE_LOCATION
+
+# Install base system requirements
+RUN sudo apt-get update \
+    && xargs sudo  apt-get install -y  < .devcontainer/requirements_system_dev.txt \
+    && sudo apt-get clean \
+    && sudo apt-get autoremove
+
+RUN sudo python3 -m venv $VIRTUAL_ENV && \
+    sudo chown -R $LOCAL_USER:$LOCAL_USER ${VIRTUAL_ENV} && \
+    . $VIRTUAL_ENV/bin/activate && \
+    python3 -m pip install --upgrade setuptools wheel && \
+    python3 -m pip install -r $FILE_LOCATION/requirements-dev.txt
```

Comment:
This is very neat and we appreciate the contribution.

Sage raises an interesting point. Devs have specific plugin/extension/python package requirements for their work. What does the workflow look like for a dev who needs to customize this? Can you create multiple development containers in a repo or are you limited to just one?

Reply:
I have never done this myself, but it seems a bit more advanced configuration is possible using Docker compose-files as described [here](https://code.visualstudio.com/remote/advancedcontainers/connect-multiple-containers#:~:text=Currently%20you%20can%20only%20connect,you%20can%20create%20separate%20devcontainer.).

A less complex solution would be adding scripts to setup the dev environment for different use cases. These scripts would then be invoked manually by the user.

As I'm writing this, maybe it would be easier to follow the thoughts of @sagesmith-wf and setup a full dev environment?

Reply:
A full dev environment for all users isn’t possible. There are folks developing private Arelle plugins.

While I appreciate the effort to help other devs onboard faster, I don't think maintaining IDE specific dev configurations is something we want to take on. I would vote in favor of a generic dev docker image though (especially if it worked with the GUI).

Reply:
That's a pity. Personally, I don't lock back to the time when I had to go through the hassle of installing a virtual environment per project I was working on, including all system dependencies. Thank you for considering it though!